lib/active_merchant/billing/gateways/fat_zebra.rb

```diff
@@ -141,23 +152,25 @@ def commit(method, uri, parameters=nil)
           message_from(response),
           response,
           :test => response['test'],
-          :authorization => authorization_from(response, success)
+          :authorization => authorization_from(response, success, uri)
         )
       end
 
       def success_from(response)
         (
           response['successful'] &&
           response['response'] &&
-          (response['response']['successful'] || response['response']['token'])
+          (response['response']['successful'] || response['response']['token'] || response['response']['response_code'] == '00')
         )
       end
 
-      def authorization_from(response, success)
+      def authorization_from(response, success, uri)
+        endpoint = uri.split('/')[0]
         if success
-          (response['response']['id'] || response['response']['token'])
+          id = response['response']['id'] || response['response']['token']
+          "#{id}|#{endpoint}"
         else
-          nil
+          ''
```

Comment:
Not something you have to change. I guess you could avoid the `else` and then call `authorization_from(response, success, uri).to_s`. Probably better to be explicit.

Reply:
calling `.to_s` would produce authorizations of the form `|purchases` etc. But in any case, now that we have `.split` being called on authorizations, it will choke on `nil`. But I suppose I could call `.to_s.split`...

Reply:
Not sure it's worth the effort, think I'd leave it as is for explicitness.

Reply:
Actually noticed a lot of split-auth implementations use `.to_s.split`, it is more robust at least. I updated that and also some variable naming conventions, if you wouldn't mind giving another once-over.

Reply:
Oops, missed amending somehow, should be accurate now.